Directory.Build.targets

```diff
@@ -0,0 +1,31 @@
+<Project>
+  <!-- 
+    The convention for names of Azure DevOps repositories mirrored from GitHub is "{GitHub org name}-{GitHub repository name}"
+  -->
+  <PropertyGroup>
+    <!-- There are quite a few git repo forms:
+      https://dnceng@dev.azure.com/dnceng/internal/_git/dotnet-arcade-services
+      https://dev.azure.com/dnceng/internal/_git/dotnet-arcade-services
+      https://dnceng.visualstudio.com/internal/_git/dotnet-arcade-services
+      dnceng@vs-ssh.visualstudio.com:v3/dnceng/internal/dotnet-arcade-services
+      git@ssh.dev.azure.com:v3/dnceng/internal/dotnet-arcade-services
+    -->
+    <_TranslateUrlPattern>(https://dnceng%40dev\.azure\.com/dnceng/internal/_git|https://dev\.azure\.com/dnceng/internal/_git|https://dnceng\.visualstudio\.com/internal/_git|dnceng%40vs-ssh\.visualstudio\.com:v3/dnceng/internal|git%40ssh\.dev\.azure\.com:v3/dnceng/internal)/([^/-]+)-(.+)</_TranslateUrlPattern>
```

Comment:
Is "%40" how we really specify "@" in our Regex? I copied the patterns from [here](https://github.com/dotnet/arcade/blob/5b4c4ebc866439f7479f17db08ad58e88e653cf0/src/Microsoft.DotNet.Arcade.Sdk/tools/RepositoryInfo.targets#L46)

Reply:
I believe we need to escape it here, because of xml parsing. That's why we use `%40`

Reply:
Hmm interesting, I guess I expected to see `\@`. Maybe both work? Good to know